firmware/src/main.cpp

```diff
@@ -1,67 +1,115 @@
+#include <pico/stdlib.h> // for uart printing
 #include <cstring>
+#include <stdio.h>
+#include <config.h>
+#include <pico/multicore.h>
 #include <harp_c_app.h>
 #include <harp_synchronizer.h>
 #include <core_registers.h>
 #include <reg_types.h>
-#include <bme688.h>
-#include <config.h>
-#ifdef DEBUG
-    #include <pico/stdlib.h> // for uart printing
-    #include <cstdio> // for printf
-#endif
+#include "core1_sensor.h"
 
 // Create device name array.
-const uint16_t who_am_i = 1234;
+const uint16_t who_am_i = 1405;
 const uint8_t hw_version_major = 1;
 const uint8_t hw_version_minor = 0;
 const uint8_t assembly_version = 2;
-const uint8_t harp_version_major = 2;
+const uint8_t harp_version_major = 0;
 const uint8_t harp_version_minor = 0;
-const uint8_t fw_version_major = 3;
-const uint8_t fw_version_minor = 0;
+const uint8_t fw_version_major = 0;
+const uint8_t fw_version_minor = 1;
 const uint16_t serial_number = ENV_SENSOR_DEVICE_ID;
 
+
+void set_led_state(bool enabled)
+{
+    if (enabled)
+    {
+        gpio_init(LED_PIN);
+        gpio_set_dir(LED_PIN, true); // true for output
+        gpio_put(LED_PIN, 0);
+    }
+    else
+        gpio_deinit(LED_PIN);
+}
+
+
+queue_t sensor_queue;
+queue_t cmd_queue;
+
+//////// Harp stuff
+
 // Harp App Register Setup.
-const size_t reg_count = 3;
+const size_t reg_count = 5;
 
 // Define register contents.
 #pragma pack(push, 1)
 struct app_regs_t
 {
-    uint16_t pressure_kpa;  // app register 0
-    uint16_t temperature_c;         // app register 1
-    uint16_t humidity_units;         // app register 1
```

Comment:
Let's replace this with pascal (looks like Pascal according to [datasheet](https://www.bosch-sensortec.com/media/boschsensortec/downloads/datasheets/bst-bme688-ds000.pdf) page 23) or whatever unit it actually is. Let's do the same with the read/write harp callback functions.